connectors/rsk.go

```diff
@@ -243,7 +243,7 @@ func (rsk *RSK) GasPrice() (uint64, error) {
 		defer cancel()
 		var price *big.Int
 		price, err = rsk.c.SuggestGasPrice(ctx)
-		if price != nil && price.Cmp(big.NewInt(0)) > 0 {
+		if price != nil && price.Cmp(big.NewInt(0)) >= 0 {
```

Comment:
I think 0 gas price should be only allowed when the network is RegTest, otherwise you could be starting the service under  unhealthy conditions.

Reply:
I am not sure in which cases gas price can be 0, but this looks fine to me. Especially taking into account that this function is only used to verify the RSK connectivity.

Reply:
0 is being returned in `regtest`

Reply:
this function is not only used to verify RSK connectivity. It's also used by the getQuoteHandler(). This means that if the node is faulty and gives 0 gas price, the quote price can be set to 0, which is an undesired effect. I'd change the function that is used for verifying the connectivity, to something like "eth_getChainId" or smth like that.

Reply:
Oh, I didn't see the use in server.go. 

The question is: can the node return 0 gas price in cases where the gas price is not actually 0? Wouldn't the node raise an error in these cases? In other words, can the node return a wrong gas price?

Reply:
Not sure how likely to happen in mainnet, but in testnet it happened a few times within 2 months.

Reply:
well, for me `0` is a valid gas price. In essence, it all depends on the node's configs, for instance, there's `minGasPrice` config for a miner. In `regtest` it's zero, that's why `eth_gasPrice` returns zero. Nothing prevents you from spinning up your own test network with this config set to zero.

Reply:
@aguschirico did it return 0 because gas price was actually 0 or because there was an error computing gas price?

Reply:
`gasPrice` is a dynamic variable and depends on a few factors including aforementioned config, gas prices of txs of recent blocks etc., and is cached in RAM, so after restarting a node, another value may be returned

Reply:
Then I think that 0 should be valid as well

Reply:
Despite the technical reasons for this value to be 0 eventually: do we want to allow a quote to be generated with gasPrice 0 in mainnet? I don't think we want, so I'd verify connectivity to the node using a different function and/or allow this value only for any env != mainnet

Reply:
But is mainnet ever going to return 0 gas price? According to this: https://stackoverflow.com/questions/55431745/how-do-geth-estimategas-and-suggest-gas-price-work

the fallback value is 1 (at least on Ethereum). Maybe it's ok to trust the value returned by the node. Otherwise, how can we tell if the gas price is valid? Could be greater than 0 and still be inaccurate.

I agree with changing the connectivity check though.

Reply:
> I agree with changing the connectivity check though.

I like this idea. I see `chainId` in the `config.json` file (under the `provider` section). Do you think we could trigger `eth_chainId` on an RSK node and compare the result with a value in the config and see if a proper type of the rsk node is connected?

Reply:
please check [this](https://github.com/rsksmart/liquidity-provider-server/pull/35/commits/d6081276c03cd6d431ecd468b0240ca382f65342) out. Used chainId endpoint for connection verification (instead of gasPrice)

Reply:
LGTM

Reply:
We could also remove chain ID from the config file if we can get it from the node.